### PR TITLE
Enhance Google Cloud org script to walk all folders in an org

### PR DESCRIPTION
**Pull request by austinkelleher**
Example usage:

```bash
yarn jupiterone-organization-setup \
  --google-access-token $(gcloud auth print-access-token) \
  --organization-id 1111111111 \
  --jupiterone-account-id MY_JUPITERONE_ACCOUNT_ID_HERE \
  --jupiterone-api-key MY_JUPITERONE_API_KEY_HERE
```

### CODE REVIEW

File: docs/jupiterone.md [hunk 1]
```diff
@@ -125,17 +125,32 @@ For more info, run any command with the `--help` flag:
 Options:
   --jupiterone-account-id <jupiteroneAccountId>  (Required) JupiterOne Account ID
   --jupiterone-api-key <jupiteroneApiKey>        (Required) JupiterOne API Key
-  --google-access-token <googleAccessToken>      (Required) Google Cloud Access Token
-  --project-id [projectId]                       (Optional) Project ID to create integration instances with. Passing multiple `--project-id` flags allows for ingestion of multiple projects.
-  -h, --help                                     Display this message
+  --google-access-token <googleAccessToken>      (Required) JupiterOne API Key
```
Comment (ndowmon) [inline docs/jupiterone.md:128]: Did we talk about this before? is this supposed to be `JupiterOne API Key`?

Comment (austinkelleher) [reply docs/jupiterone.md]: Whoops...fixed.

File: docs/jupiterone.md [hunk 2]
```diff
@@ -125,17 +125,32 @@ For more info, run any command with the `--help` flag:
...
+  --organization-id [organizationId]             (Optional) Array of organization IDs to collect projects from
+  --project-id [projectId]                       (Optional) Array of project IDs to create integration instances with
+  --skip-system-projects [skipSystemProjects]    (Optional) Skips creation of any projects that have an ID that start with "sys-" (default: false)
+  -h, --help                                     Display this messagee
```
Comment (a-u-h-g) [inline docs/jupiterone.md:132]: message*

Comment (austinkelleher) [reply docs/jupiterone.md]: Thanks!

**Review by ndowmon — DISMISSED** [review]

**Review by aiwilliams — APPROVED** [review]
🔥 

What do you think about adding a `--dry-run` option and just listing what would be done?

File: docs/jupiterone.md [hunk 3]
```diff
@@ -126,16 +126,31 @@ Options:
...
 ```sh
 yarn jupiterone-organization-setup \
   --google-access-token $(gcloud auth print-access-token) \
+  --organization-id 1111111111 \
+  --jupiterone-account-id MY_JUPITERONE_ACCOUNT_ID_HERE \
+  --jupiterone-api-key MY_JUPITERONE_API_KEY_HERE
+```
+
+Example usage to create integration instances for every project that are under a
+multiple Google Cloud organizations:
```
Comment (aiwilliams) [inline docs/jupiterone.md:147]:
```suggestion
Example usage to create integration instances for each project in multiple
Google Cloud organizations:
```

File: scripts/organizationSetup.ts [hunk 4]
```diff
@@ -279,62 +368,69 @@ async function createJupiterOneIntegrationInstance(
...
+              serviceAccountKeyFile: serviceAccountKey,
             },
-            serviceAccountKeyFile: serviceAccountKey,
+            integrationDefinitionId: GOOGLE_CLOUD_INTEGRATION_DEFINITION_ID,
           },
-          integrationDefinitionId: GOOGLE_CLOUD_INTEGRATION_DEFINITION_ID,
         },
+      }),
+      headers: {
+        'LifeOmic-Account': jupiteroneAccountId,
```
Comment (aiwilliams) [inline scripts/organizationSetup.ts:395]: Should we move to the newly supported `JupiterOne-Account` so we have one less place to change it in the future? 😄

**Review by aiwilliams — APPROVED** [review]

**Review by aiwilliams — APPROVED** [review]

**Review by ndowmon — APPROVED** [review]

File: scripts/organizationSetup.ts [hunk 5]
```diff
@@ -401,17 +498,38 @@ async function setupOrganizationProject(
...
+  return SetupOrganizationProjectResult.CREATED;
 }
 
-export async function setupOrganization(params: SetupOrganizationParams) {
+export interface SetupOrganizationResult {
+  created: string[];
+  failed: string[];
+  skipped: string[];
+  exists: string[];
+}
```
Comment (mknoedel) [inline scripts/organizationSetup.ts:510]: Super unimportant, but you can do a cool type here if you are ok with having these keys be uppercase
`type SetupOrganizationResult = {
  [key in SetupOrganizationProjectResult]: string[]
}`

And if we upgrade to Typescript 4.1 you could keep the lowercase with:
```
   type SetupOrganizationResult = {
     [key in `${lowercase SetupOrganizationProjectResult}`]: string[]
   }
```
which is kind of fun

Comment (aiwilliams) [reply scripts/organizationSetup.ts:510]: That is cool, I don't think I spend enough time studying type generation feature like that!

**Pull request merged**